src/utils/style-utils.js

```diff
@@ -90,6 +90,7 @@ export const setTextFontSize = (text, font, textFontSize, hasIcon) => {
 
 const setFontSizeBehaviorStyle = (button, text, textSpan) => {
   text.offsetWidth <= button.clientWidth ? (textSpan.style.overflow = "visible") : (textSpan.style.overflow = "hidden");
+  text.children[0].style.marginRight = `${trimDecimal(text.offsetWidth * 0.04)}px`;
```

Comment:
What happens when there is no icon selected?

Reply:
It is displayed this way when no icon set. 
![image](https://github.com/qlik-oss/sn-action-button/assets/91126632/41e44cf3-f14c-40a3-9db6-dd89869476e6)

Reply:
But it sets a marginRight on the text instead then? I guess that is why all the rendering tests are complaining

Reply:
It should not, because it only sets the margin for the first child which is the icon.

Reply:
Yes, but when you do not have an icon selected the first child would be the text instead right?

Reply:
Indeed, a hasIcon check should be there. REverting the rendering tests first and then adding the check.

Reply:
also, isn't there a setting to have the icon to the right of the text instead of to the left?

Reply:
Will check those options too. Thanks for pointing out.

Reply:
@eliseeborn , checked all the combo. Seems work. Now need to see if rendering tests need fix, since there has been a little UI change with the space. 
![image](https://github.com/qlik-oss/sn-action-button/assets/91126632/ac08003a-fdae-43b0-930a-dbc422da24c7)